src/main/java/com/cloudbees/jenkins/support/startup/ShutdownComponent.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2025 CloudBees, Inc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.support.startup;
+
+import com.cloudbees.jenkins.support.SupportPlugin;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.impl.ThreadDumps;
+import com.cloudbees.jenkins.support.timer.FileListCap;
+import com.cloudbees.jenkins.support.timer.UnfilteredFileListCapComponent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.model.listeners.ItemListener;
+import hudson.util.ClassLoaderSanityThreadFactory;
+import hudson.util.DaemonThreadFactory;
+import hudson.util.ExceptionCatchingThreadFactory;
+import hudson.util.NamingThreadFactory;
+import java.io.File;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.util.SystemProperties;
+
+/**
+ * Collects thread dumps during shutdown if it exceeds a certain threshold.
+ */
+@Extension
+public class ShutdownComponent extends UnfilteredFileListCapComponent {
+    private static final Logger LOGGER = Logger.getLogger(ShutdownComponent.class.getName());
+
+    private static final int THREAD_DUMPS_TO_RETAIN =
+            SystemProperties.getInteger(ShutdownComponent.class.getName() + ".THREAD_DUMPS_TO_RETAIN", 40);
+    private static final int TOTAL_ITERATIONS =
+            SystemProperties.getInteger(ShutdownComponent.class.getName() + ".TOTAL_ITERATIONS", 4);
+    private static final int DELAY_BETWEEN_THREAD_DUMPS_MS =
+            SystemProperties.getInteger(ShutdownComponent.class.getName() + ".DELAY_BETWEEN_THREAD_DUMPS_MS", 1000);
```

Comment:
I think we should introduce a method in `SystemProperties` to get a `Duration` where you can specify a default time unit but the parser would also accept explicit time units like `2s` etc.

Reply:
A while ago I introduced https://github.com/jenkinsci/remoting/blob/3ea58fc6d3356698eb8b14210135149b8f1e6018/src/main/java/org/jenkinsci/remoting/util/DurationStyle.java#L38-L39 in remoting. Could unrestrict it so that Jenkins core can facade it through `SystemProperties`

Reply:
https://github.com/jenkinsci/remoting/pull/800 + https://github.com/jenkinsci/jenkins/pull/10561